dj_hetmech_app/views.py

```diff
@@ -187,8 +177,14 @@ def get(self, request):
                 target_node.metanode,
                 extra_filters=~Q(abbreviation__in=metapaths_present),
             )
+            if limit is not None:
+                metapath_qs = metapath_qs[:limit - len(pathcounts)]
             pathcounts += MetapathSerializer(metapath_qs, many=True).data
 
+        # `metapath_qs = metapath_qs[:0]` doesn't filter to an empty query set`
+        if limit is not None:
+            pathcounts = pathcounts[:limit]
```

Comment:
170-171 truncate pathcounts. However then `if complete` logic potentially extends pathcounts. If 180-181 worked, then it would only extend pathcounts up to limit. But it doesn't work, a despicable django bug IMO. Thus we have to truncate here again.